artemis/modules/domain_scanner.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+from karton.core import Task
+from artemis.resolvers import lookup, ResolutionException
+
+from artemis.binds import TaskStatus, TaskType
+from artemis.module_base import ArtemisBase
+
+class DomainScanner(ArtemisBase):
+    """
+    DNS checker to verify if domains exist in the DNS system.
+    """
+
+    identity = "domain_scanner"
+    filters = [
+        {"type": TaskType.DOMAIN.value},
+    ]
+
+    def run(self, current_task: Task) -> None:
+        domain = current_task.get_payload(TaskType.DOMAIN)
+        domain_exists = self.check_domain_exists(domain)
+
+        if domain_exists:
+            self.db.save_task_result(
+                task=current_task,
+                status=TaskStatus.OK,
+                data={
+                    "domain": domain,
+                    "exists": True
+                }
+            )
+            # Create a new task for the next module
+            new_task = Task(
+                {"type": TaskType.DOMAIN},
```

Comment:
how do you plan to make other modules distinguish between filtered and unfiltered domains? maybe create a new task type for unfiltered domain and keep domain for filtered?